server/api_archive_integration_test.go

```diff
@@ -0,0 +1,275 @@
+package server
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"os"
+	"path/filepath"
+	"slices"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/glowlabs-org/gca-backend/glow"
+)
+
+func readFile(dir, name string) ([]byte, error) {
+	path := filepath.Join(dir, name)
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func TestApiArchive(t *testing.T) {
+	// Create a populated test environment and start a new server.
+	gcas, dir, err := ServerTestEnvironment(t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer gcas.Close()
+
+	fileMap := map[string]bool{}
+	dataMap := map[string][]byte{}
+
+	// Add the public files to the test.
+	for _, name := range PublicFiles() {
+		data, err := readFile(dir, name)
+		if err != nil {
+			t.Fatal(err)
+		}
+		fileMap[name] = false
+		dataMap[name] = data
+	}
+
+	// Add the public key file
+	const pkf = "server.pubkey"
+	fileMap[pkf] = false
+	data, err := readFile(dir, "server.keys") // server.pubkey should be the first 32 bytes from server.keys
+	if err != nil {
+		t.Fatal(err)
+	}
+	dataMap[pkf] = data[:32]
+
+	const rmf = "README"
+	fileMap[rmf] = false
+	dataMap[rmf] = []byte(ReadmeContents)
+
+	// Post the archive request
+	resp, err := http.Get(fmt.Sprintf("http://localhost:%v/api/v1/archive", gcas.httpPort))
+	if err != nil {
+		t.Fatal(err)
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		t.Fatal(fmt.Errorf("expected status 200, but got %d: %s", resp.StatusCode, string(body)))
+	}
+
+	reader := bytes.NewReader(body)
+
+	rzip, err := zip.NewReader(reader, int64(len(body)))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, zipf := range rzip.File {
+		// Make sure there is not an extra file in the zip contents,
+		// and keep track of the ones we find.
+		if _, ok := fileMap[zipf.Name]; ok {
+			fileMap[zipf.Name] = true
+		} else {
+			t.Fatal(fmt.Errorf("%v from zipfile is not a valid file", zipf.Name))
+		}
+
+		// Check each file against the original data.
+
+		f, err := zipf.Open()
+		if err != nil {
+			t.Fatal(fmt.Errorf("%v from zipfile could not be opened: %v", zipf.Name, err))
+		}
+		defer f.Close()
+
+		data, err := io.ReadAll(f)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if zipf.Name == "server.keys" {
+			// Special case, must have the public key as the first 32 bytes,
+			// followed by 64 zeroes.
+
+			if len(data) != 96 || len(data) != len(dataMap[zipf.Name]) {
+				t.Fatal(fmt.Errorf("%v from zipfile wrong length", zipf.Name))
+			}
+			if !bytes.Equal(data[:32], dataMap[zipf.Name][:32]) {
+				t.Fatal(fmt.Errorf("file %v data does not match original file", zipf.Name))
+			}
+			zbuf := make([]byte, 64)
+			if !bytes.Equal(data[32:], zbuf) {
```

Comment:
an alternative option that might improve readability would be to zero out the data after you save it in the map

Reply:
This code is no longer needed and was removed.